### ISSUE TITLE
Implement BERT model in TTML and benchmark against PyTorch and TTNN implementations

### ISSUE THREAD

**Issue by aliaksei-sala** [tenstorrent/tt-metal#26739]
**Description**

Implement a BERT model in TTML based on existing model implementations, such as:
[llama.cpp](https://github.com/tenstorrent/tt-metal/blob/main/tt-train/sources/ttml/models/distributed/llama.cpp)

**Requirements:**

- Focus on inference. If any required operations for backward propagation are missing, throw an exception with a fully descriptive error message.
- Benchmark and compare the performance of the BERT model when run using:
  - [PyTorch 2.0 TTNN](https://github.com/tenstorrent/pytorch2.0_ttnn) in eval mode
  - [TTNN model](https://github.com/tenstorrent/tt-metal/models/demos/metal_BERT_large_11/README.md)
  - TTML model (Implemented as part of this GitHub issue)
- Benchmark results must be attached to the PR description.
  For the TTML implementation, include additional profiling information to help identify bottlenecks(if any).
- Set default sequence length to 1024.

**Comment by dmakoviichuk-tt** [comment]
A few more details.
In the tt-train project you need:
1) create one more example project: inference_demo
2) create a new model called Bert in the ttml::models (same as llama and gpt2). My expectation new ops are not needed.
3) By this time I should merge safetensors format support. So you can write import from hugging face.
4) Run measurements on different batch sizes and compare with other solutions.
